src/viewer/Viewer.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { Component } from 'react'
+import classNames from 'classnames'
+
+import ViewerToolbar from './ViewerToolbar'
+import ImageViewer from './ImageViewer'
+import NoViewer from './NoViewer'
+
+import styles from './viewer'
+
+const KEY_CODE_LEFT = 37
+const KEY_CODE_RIGHT = 39
+
+export default class Viewer extends Component {
+  state = {
+    controlsHidden: false
+  }
+
+  showControls = () => {
+    this.setState({ controlsHidden: false })
+    document.removeEventListener('mousemove', this.showControls)
+  }
+  hideControls = () => {
+    this.setState({ controlsHidden: true })
+    document.addEventListener('mousemove', this.showControls)
+  }
+
+  componentDidMount() {
+    document.addEventListener('keydown', this.onKeyDown, false)
+    document.addEventListener('mousemove', this.showControls)
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this.onKeyDownCallback, false)
+    document.removeEventListener('mousemove', this.showControls)
+  }
+
+  onKeyDown = e => {
+    if (e.keyCode === KEY_CODE_LEFT) this.onPrevious()
+    else if (e.keyCode === KEY_CODE_RIGHT) this.onNext()
+  }
+
+  onNext = () => {
+    const { files, currentIndex } = this.props
+    if (currentIndex === files.length - 1) {
+      return
+    }
+    const nextIndex = currentIndex + 1
+    const nextFile = files[nextIndex]
+    this.onChange(nextFile, nextIndex)
+  }
+
+  onPrevious = () => {
+    const { files, currentIndex } = this.props
+    if (currentIndex === 0) {
+      return
+    }
+    const prevIndex = currentIndex - 1
+    const prevFile = files[prevIndex]
+    this.onChange(prevFile, prevIndex)
+  }
+
+  onChange(nextFile, nextIndex) {
+    if (this.props.onChange) {
+      this.props.onChange(nextFile, nextIndex)
+    }
+  }
+
+  render() {
+    const { files, currentIndex, onClose } = this.props
+    const currentFile = files[currentIndex]
+    const fileCount = files.length
+    const hasPrevious = currentIndex > 0
+    const hasNext = currentIndex < fileCount - 1
```

Comment:
I may be wrong, but I believe the current behavior is to keep rotating the photos, ie. if you click "next" while on viewing the last file in a collection, is goes to the first.

Reply:
I don't think so, I'll ask the PO ;)